pkg/kubecost/allocation.go

```diff
@@ -358,28 +360,32 @@ func (a *Allocation) add(that *Allocation) {
 		log.Warningf("Allocation.AggregateBy: trying to add a nil receiver")
 		return
 	}
+	if a.Properties != nil && that.Properties != nil {
+		aCluster := a.Properties.Cluster
+		thatCluster := that.Properties.Cluster
+		aNode := a.Properties.Node
+		thatNode := that.Properties.Node
 
-	aCluster, _ := a.Properties.GetCluster()
-	thatCluster, _ := that.Properties.GetCluster()
-	aNode, _ := a.Properties.GetNode()
-	thatNode, _ := that.Properties.GetNode()
+		// reset properties
+		a.Properties = nil
 
-	// reset properties
-	a.Properties = nil
-
-	// ensure that we carry cluster ID and/or node over if they're the same
-	// required for idle/shared cost allocation
-	if aCluster == thatCluster {
-		a.Properties = Properties{ClusterProp: aCluster}
-	}
-	if aNode == thatNode {
-		if a.Properties == nil {
-			a.Properties = Properties{NodeProp: aNode}
-		} else {
-			a.Properties.SetNode(aNode)
+		// ensure that we carry cluster ID and/or node over if they're the same
+		// required for idle/shared cost allocation
+		if aCluster == thatCluster {
+			a.Properties = &AllocationProperties{Cluster: aCluster}
+		}
+		if aNode == thatNode {
+			if a.Properties == nil {
+				a.Properties = &AllocationProperties{Node: aNode}
+			} else {
+				a.Properties.Node = aNode
+			}
 		}
+	} else {
+		a.Properties = nil
 	}
```

Comment:
This, we'll want to discuss. We only do this because of the old, terrible, map implementation.